subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/matcher/RecyclerViewMatcher.kt

```diff
@@ -48,6 +49,15 @@ class RecyclerViewMatcher {
         }
     }
 
+    fun hasViewTypeAtPosition(position: Int, viewType: Int): Matcher<View> = object : RecyclerViewMatcher() {
+        override fun performMatching(recyclerView: RecyclerView): Boolean {
+            val adapter = recyclerView.adapter ?: error("Adapter is not attached to RecyclerView")
```

Comment:
requireNotNull(recyclerView.adapter) { "Adapter is not attached to RecyclerView" }